lib/core/sync_job_runner.rb

```diff
@@ -49,7 +56,11 @@ def execute
     rescue StandardError => e
       @status[:error] = e.message
     ensure
-      ElasticConnectorActions.complete_sync(@connector_settings.id, job_id, @status.dup)
+      if job_id.present?
```

Comment:
It's failing if there's no job claimed. This change is actually already in [another branch](https://github.com/elastic/connectors-ruby/pull/174/files#diff-ff8974a0d605a7d8f2dc0e25b718febce7372e71dbf57aaf02a9ae1de9e0457dR65), but I had to do it for testing purposes, since otherwise the code just fails when it's not finding a scheduled job (which is a normal situation).